Library/Homebrew/test/gpg_spec.rb

```diff
@@ -12,8 +12,15 @@
 
         shutup do
           subject.create_test_key(dir)
+          gpg = subject::GPG_EXECUTABLE
+          @version = Utils.popen_read(gpg, "--version")[/\d\.\d/, 0]
```

Comment:
Alternatively now I'm awake and thinking clearer: mock/stub the methods accordingly so it only ever hits one path and then add two tests. Shout if you need more help with that.